render-to-string.ts

```diff
@@ -0,0 +1,60 @@
+import mdx, { Options } from '@mdx-js/mdx'
+import { MDXProvider, mdx as mdxReact } from '@mdx-js/react'
+import { transformAsync } from '@babel/core'
+import presetEnv from '@babel/preset-env'
+import presetReact from '@babel/preset-react'
+import { renderToString as reactRenderToString } from 'react-dom/server'
+import React from 'react'
+// @ts-ignore
+import pluginBrowser from './babel-plugin-mdx-browser'
+import { Components, Scope, Source } from './types'
+
+export default async function renderToString(
+  source: string,
+  components?: Components,
+  options?: Options,
+  scope: Scope = {}
+): Promise<Source> {
+  // transform it into react
+  const code = await mdx(source, { ...options, skipExport: true })
+  const [now, later] = await Promise.all([
+    // this one is for immediate evaluation so we can renderToString below
+    transformAsync(code, {
+      presets: [presetReact, presetEnv],
+      configFile: false,
+    }),
+    // this one is for the browser to eval and rehydrate, later
+    transformAsync(code, {
+      presets: [presetReact, presetEnv],
+      plugins: [pluginBrowser],
+      configFile: false,
+    }),
+  ])
+
+  if (!now || !later || !later.code) {
+    throw new Error('Failed to transform mdx source code')
+  }
```

Comment:
This doesn't seem like a super useful error message - do you have any more context on when you encountered this and what caused it? Maybe we can add more detail to the error?

Reply:
The return type of `transformAsync` is `Promise<BabelFileResult | null>` and we need to handle the null case. Which is unlikely to happen since it only depends on babel plugins and `code` const which is returned by `mdx(...)`.